### ISSUE TITLE
Deleting objects does not work

### ISSUE THREAD

**Issue by FinnStutzenstein** [OpenSlides/openslides-autoupdate-service#99]
With the following requests `topic/1` (see example data) is deleted.

```
{
 "collection": "meeting",
 "ids": [1],
 "fields": {
  "id": null,
  "agenda_item_ids": {
   "type": "relation-list",
   "collection": "agenda_item",
   "fields": {
    "item_number": null,
    "comment": null,
    "closed": null,
    "type": null,
    "is_hidden": null,
    "is_internal": null,
    "duration": null,
    "weight": null,
    "level": null,
    "parent_id": null,
    "child_ids": null,
    "meeting_id": null,
    "id": null,
    "content_object_id": {
     "type": "generic-relation",
     "fields": {
      "title": null,
      "number": null,
      "id": null,
      "list_of_speakers_id": {
       "type": "relation",
       "collection": "list_of_speakers",
       "fields": {
        "closed": null,
        "id": null,
        "speaker_ids": {
         "type": "relation-list",
         "collection": "speaker",
         "fields": {
          "begin_time": null,
          "end_time": null,
          "id": null
         }
        }
       }
      }
     }
    }
   }
  }
 }
}
```

 This is the current write request from the backend (which is also wrong, but not relevant for this bug).

```
{'events': [{'type': 'delete', 'fqid': FQId topic/1}, {'type': 'update', 'fqid': FQId agenda_item/3, 'fields': {'content_object_id': None}}, {'type': 'update', 'fqid': FQId meeting/1, 'fields': {'topic_ids': [2, 3, 4, 5, 6, 7, 8]}}], 'information': {FQId topic/1: ['Object deleted'], FQId agenda_item/3: ['Object attachment to topic reset'], FQId meeting/1: ['Object attachment to topic reset']}, 'user_id': 1}
```

Mainly: agenda_item/3/content_object_id -> None, topic/1 deleted.

Note: The request was not active when the deletion occured (since for it I went into the topic-detail view). After the deletion I'm redirected to the list view with the request. There I get the same autoupdate as before, also when reloading, so the agenda_item/3 has still the content_object_id and topic/1 exists. When I restart the autoupdate service (`./dc-dev.sh restart autoupdate`) I receive a correct autoupdate.

**Comment by ostcar** [comment]
Could you paste the redis xadd command, that is "send" from the datastore to the autoupdate service?

**Comment by FinnStutzenstein** [comment]
"XADD" "ModifiedFields" "*" "topic/1/id" "null" "topic/1/text" "null" "topic/1/title" "null" "topic/1/tag_ids" "null" "topic/1/meeting_id" "null" "topic/1/meta_deleted" "null" "topic/1/agenda_item_id" "null" "topic/1/attachment_ids" "null" "topic/1/list_of_speakers_id" "null" "topic/1/meta_position" "2" "agenda_item/3/content_object_id" "null" "agenda_item/3/meta_position" "2" "meeting/1/topic_ids" "[2, 3, 4, 5, 6, 7, 8]" "meeting/1/meta_position" "2"


The only thing I suspect to be wrong is `topic/1/meta_deleted`. It should be False, shouldn't it?

**Comment by ostcar** [comment]
It should be null but I don't think, this is the Problem. The autoupdate service works on keys and values, so there is no connection between the keys. If one key is wrong, it should not affect the other keys.

Could test it with curl? I mean create a curl request that requests the date, then send the redis xadd command and afterwards show me the output and the expected output.

**Comment by FinnStutzenstein** [comment]
Curl
```
curl -Nk https://localhost:9012/system/autoupdate -d '[{"collection":"meeting","ids":[1],"fields":{"id":null,"agenda_item_ids":{"type":"relation-list","collection":"agenda_item","fields":{"item_number":null,"comment":null,"closed":null,"type":null,"is_hidden":null,"is_internal":null,"duration":null,"weight":null,"level":null,"parent_id":null,"child_ids":null,"meeting_id":null,"id":null,"content_object_id":{"type":"generic-relation","fields":{"title":null,"number":null,"id":null,"list_of_speakers_id":{"type":"relation","collection":"list_of_speakers","fields":{"closed":null,"id":null,"speaker_ids":{"type":"relation-list","collection":"speaker","fields":{"begin_time":null,"end_time":null,"id":null}}}}}}}}}}]'
```
Initial Autpupdate:
```
{"list_of_speakers/15/id":15,"agenda_item/6/id":6,"agenda_item/11/content_object_id":"assignment/1","agenda_item/2/closed":false,"agenda_item/1/child_ids":[],"agenda_item/5/is_internal":true,"topic/7/id":7,"list_of_speakers/7/id":7,"list_of_speakers/4/speaker_ids":[],"agenda_item/14/level":1,"agenda_item/6/closed":false,"list_of_speakers/11/speaker_ids":[6,7,8],"speaker/10/id":10,"speaker/4/id":4,"agenda_item/4/closed":false,"agenda_item/12/type":2,"motion/2/id":2,"list_of_speakers/12/speaker_ids":[],"list_of_speakers/9/closed":false,"list_of_speakers/5/id":5,"agenda_item/7/type":1,"agenda_item/1/level":0,"agenda_item/7/child_ids":[],"list_of_speakers/5/closed":false,"topic/2/list_of_speakers_id":4,"motion/3/list_of_speakers_id":12,"agenda_item/1/is_internal":false,"list_of_speakers/14/id":14,"speaker/9/id":9,"agenda_item/11/meeting_id":1,"agenda_item/14/meeting_id":1,"agenda_item/13/child_ids":[],"agenda_item/8/closed":false,"agenda_item/1/meeting_id":1,"agenda_item/2/child_ids":[],"agenda_item/2/is_internal":true,"assignment/2/title":"2. Wahl","motion/1/number":"A1","agenda_item/4/weight":1001,"agenda_item/15/child_ids":[],"agenda_item/7/meeting_id":1,"agenda_item/13/meeting_id":1,"agenda_item/3/item_number":"","motion/4/id":4,"list_of_speakers/10/closed":false,"speaker/7/id":7,"agenda_item/11/is_hidden":false,"agenda_item/14/parent_id":3,"agenda_item/1/content_object_id":"motion/1","assignment/2/list_of_speakers_id":14,"motion/3/id":3,"list_of_speakers/3/id":3,"agenda_item/15/is_internal":true,"agenda_item/8/type":1,"agenda_item/11/level":0,"agenda_item/10/is_internal":false,"agenda_item/2/meeting_id":1,"list_of_speakers/3/closed":false,"list_of_speakers/2/closed":false,"list_of_speakers/11/id":11,"agenda_item/15/weight":10000,"agenda_item/13/level":0,"agenda_item/2/level":0,"topic/7/list_of_speakers_id":9,"agenda_item/2/is_hidden":false,"agenda_item/12/meeting_id":1,"agenda_item/14/is_hidden":false,"agenda_item/3/is_hidden":false,"agenda_item/3/closed":false,"agenda_item/9/child_ids":[],"motion/3/title":"ohne","list_of_speakers/13/speaker_ids":[],"topic/5/title":"E","topic/7/title":"G","list_of_speakers/4/closed":false,"agenda_item/15/id":15,"agenda_item/6/item_number":"","agenda_item/11/is_internal":true,"agenda_item/8/item_number":"","agenda_item/9/is_internal":false,"agenda_item/5/item_number":"","agenda_item/14/child_ids":[],"agenda_item/6/level":0,"topic/6/id":6,"speaker/13/id":13,"agenda_item/1/closed":false,"agenda_item/3/meeting_id":1,"agenda_item/5/child_ids":[],"agenda_item/9/content_object_id":"topic/7","agenda_item/14/item_number":"","list_of_speakers/5/speaker_ids":[],"agenda_item/13/is_internal":true,"agenda_item/3/id":3,"agenda_item/12/weight":10000,"list_of_speakers/4/id":4,"agenda_item/10/weight":1007,"agenda_item/14/type":2,"motion/4/number":"3","topic/1/list_of_speakers_id":3,"list_of_speakers/13/id":13,"agenda_item/1/is_hidden":false,"topic/6/list_of_speakers_id":8,"agenda_item/7/weight":1004,"agenda_item/8/meeting_id":1,"agenda_item/13/is_hidden":false,"agenda_item/8/level":0,"agenda_item/1/type":1,"agenda_item/5/closed":false,"agenda_item/14/is_internal":true,"agenda_item/11/weight":10000,"agenda_item/3/type":1,"agenda_item/6/content_object_id":"topic/4","agenda_item/2/type":2,"agenda_item/11/id":11,"motion/4/list_of_speakers_id":13,"motion/1/title":"test","list_of_speakers/9/speaker_ids":[],"speaker/6/id":6,"motion/1/id":1,"topic/1/id":1,"topic/3/title":"C","agenda_item/10/id":10,"agenda_item/15/type":2,"topic/3/id":3,"topic/2/id":2,"agenda_item/6/meeting_id":1,"agenda_item/6/is_internal":true,"agenda_item/8/child_ids":[],"list_of_speakers/3/speaker_ids":[1,2],"agenda_item/5/content_object_id":"topic/3","agenda_item/2/item_number":"","agenda_item/14/weight":1001,"list_of_speakers/1/speaker_ids":[11,12,13],"speaker/11/begin_time":1584512636,"motion/2/list_of_speakers_id":2,"list_of_speakers/6/id":6,"list_of_speakers/7/speaker_ids":[3,4],"agenda_item/6/type":2,"agenda_item/13/weight":10000,"agenda_item/13/item_number":"","agenda_item/12/is_internal":true,"list_of_speakers/9/id":9,"topic/5/id":5,"list_of_speakers/12/id":12,"list_of_speakers/14/closed":false,"meeting/1/id":1,"meeting/1/agenda_item_ids":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15],"agenda_item/9/meeting_id":1,"agenda_item/4/item_number":"","motion/1/list_of_speakers_id":1,"topic/4/list_of_speakers_id":6,"list_of_speakers/15/speaker_ids":[],"agenda_item/15/level":0,"agenda_item/2/id":2,"agenda_item/10/meeting_id":1,"motion/2/number":"1 - 1","agenda_item/2/content_object_id":"motion/2","agenda_item/14/content_object_id":"assignment/2","topic/4/title":"D","list_of_speakers/6/closed":false,"agenda_item/1/id":1,"list_of_speakers/7/closed":false,"agenda_item/12/closed":false,"agenda_item/7/item_number":"","agenda_item/13/type":2,"agenda_item/8/is_internal":false,"list_of_speakers/10/speaker_ids":[],"agenda_item/8/is_hidden":false,"agenda_item/2/weight":10000,"agenda_item/10/is_hidden":true,"agenda_item/9/item_number":"","agenda_item/4/id":4,"speaker/3/id":3,"agenda_item/7/content_object_id":"topic/5","motion/4/title":"komplex","list_of_speakers/15/closed":false,"speaker/12/id":12,"agenda_item/4/type":1,"agenda_item/9/is_hidden":false,"agenda_item/15/closed":false,"topic/2/title":"B","motion_block/1/title":"BLOCK A","agenda_item/3/level":0,"agenda_item/9/weight":1006,"speaker/11/end_time":1584512638,"list_of_speakers/1/closed":false,"agenda_item/4/level":0,"agenda_item/7/closed":false,"agenda_item/8/id":8,"agenda_item/12/content_object_id":"motion/3","motion_block/1/list_of_speakers_id":15,"topic/5/list_of_speakers_id":7,"agenda_item/4/is_hidden":false,"agenda_item/10/type":3,"agenda_item/12/id":12,"list_of_speakers/8/speaker_ids":[5],"agenda_item/5/level":0,"topic/4/id":4,"list_of_speakers/8/closed":false,"agenda_item/10/closed":false,"speaker/5/id":5,"agenda_item/4/child_ids":[],"agenda_item/11/closed":false,"agenda_item/7/is_internal":false,"agenda_item/13/id":13,"agenda_item/10/item_number":"","agenda_item/3/is_internal":false,"agenda_item/7/id":7,"motion_block/1/id":1,"list_of_speakers/10/id":10,"agenda_item/4/is_internal":false,"motion/3/number":"2","list_of_speakers/2/id":2,"list_of_speakers/6/speaker_ids":[],"agenda_item/9/type":1,"list_of_speakers/11/closed":false,"agenda_item/4/content_object_id":"topic/2","agenda_item/1/item_number":"","agenda_item/5/meeting_id":1,"speaker/8/id":8,"agenda_item/10/content_object_id":"topic/8","agenda_item/11/type":2,"agenda_item/15/is_hidden":false,"agenda_item/5/weight":1002,"list_of_speakers/2/speaker_ids":[],"assignment/1/list_of_speakers_id":11,"agenda_item/13/content_object_id":"motion/4","agenda_item/7/is_hidden":false,"agenda_item/12/level":0,"agenda_item/15/item_number":"","motion/2/title":"\u00c4nderungsantrag zu 1","agenda_item/9/id":9,"topic/8/title":"H","agenda_item/8/weight":1005,"agenda_item/15/meeting_id":1,"agenda_item/11/item_number":"","agenda_item/4/meeting_id":1,"agenda_item/8/content_object_id":"topic/6","agenda_item/3/content_object_id":"topic/1","speaker/1/id":1,"agenda_item/1/weight":10000,"agenda_item/5/is_hidden":false,"agenda_item/3/child_ids":[14],"topic/6/title":"F","topic/8/list_of_speakers_id":10,"topic/3/list_of_speakers_id":5,"list_of_speakers/13/closed":false,"agenda_item/9/level":0,"agenda_item/6/is_hidden":false,"agenda_item/5/id":5,"agenda_item/12/is_hidden":false,"assignment/2/id":2,"list_of_speakers/12/closed":false,"speaker/11/id":11,"agenda_item/3/weight":1000,"agenda_item/5/type":2,"agenda_item/12/item_number":"","agenda_item/15/content_object_id":"motion_block/1","agenda_item/7/level":0,"agenda_item/14/closed":false,"topic/1/title":"A","list_of_speakers/14/speaker_ids":[9,10],"agenda_item/12/child_ids":[],"agenda_item/9/closed":false,"agenda_item/6/weight":1003,"agenda_item/10/child_ids":[],"agenda_item/13/closed":false,"agenda_item/14/id":14,"agenda_item/6/child_ids":[],"topic/8/id":8,"list_of_speakers/1/id":1,"assignment/1/title":"Wahl","assignment/1/id":1,"speaker/2/id":2,"agenda_item/11/child_ids":[],"agenda_item/10/level":0,"list_of_speakers/8/id":8}
```
XADD:
```
"XADD" "ModifiedFields" "*" "topic/1/id" "null" "topic/1/text" "null" "topic/1/title" "null" "topic/1/tag_ids" "null" "topic/1/meeting_id" "null" "topic/1/meta_deleted" "null" "topic/1/agenda_item_id" "null" "topic/1/attachment_ids" "null" "topic/1/list_of_speakers_id" "null" "topic/1/meta_position" "2" "agenda_item/3/content_object_id" "null" "agenda_item/3/meta_position" "2" "meeting/1/topic_ids" "[2, 3, 4, 5, 6, 7, 8]" "meeting/1/meta_position" "2"
```
Autoupdate:
```
{"topic/1/id":null}
```

Expected: More fields ;) E.g. `agenda_item/3/content_object_id`.

**Comment by ostcar** [comment]
I found and fixed a lot of bugs looking into this issue. I hope it is correct now. After the update, the following keys are returned now:

`{"agenda_item/3/content_object_id":null,"topic/1/title":null,"topic/1/list_of_speakers_id":null,"topic/1/id":null}`

I also fixed the first output. I always thought, that the first response should not contain keys where the value is `null`. But this did not work. If the client did expect `null` values, it could fail now.
